process/economics/economicsData.go

```diff
@@ -223,6 +223,7 @@ func (ed *EconomicsData) ComputeMoveBalanceFee(tx process.TransactionWithFeeHand
 
 // ComputeTxFee computes the provided transaction's fee using enable from epoch approach
 func (ed *EconomicsData) ComputeTxFee(tx process.TransactionWithFeeHandler) *big.Int {
+	//TODO: if flag is set, always take all the fee?
```

Comment:
yes. as otherwise it is an error.

Reply:
in other components this function may be wrongly used.

Reply:
this changed, now also SCs are taken in to consideration... 

is line 258 correct? When can this happen? The transaction has a lower limit than the moveBalance?
if tx.GetGasLimit() <= gasLimitForMoveBalance {
        return moveBalanceFee
}

Reply:
this was a bug initially - as computetxfee used only movebalance fee.
this introduced errors in how the function was used. now the code is correct. It is not possible for tx to have a gaslimit lower than moveBalance gas cost. It is checked by interceptor.

Reply:
ok